src/blueapi/plugins/data_writing_server.py

```diff
@@ -0,0 +1,76 @@
+import itertools
+import os
+from pathlib import Path
+from typing import Dict
+
+from fastapi import FastAPI
+from pydantic import BaseModel
+
+app = FastAPI()
+
+SCAN_NUMBER = itertools.count()
+
+
+class DataCollection(BaseModel):
+    collection_number: int
+    group: str
+    raw_data_files_root: Path
+    nexus_file_path: Path
+
+
+class DataCollectionSetupResult(BaseModel):
+    collection: DataCollection
+    directories_created: bool
+
+
+COLLECTIONS: Dict[str, Dict[int, DataCollection]] = {}
+
+
+@app.post("/collection/{group}")
+def create_collection(group: str) -> DataCollectionSetupResult:
+    num = next(SCAN_NUMBER)
+    if group not in COLLECTIONS:
+        COLLECTIONS[group] = {}
+    if num in COLLECTIONS[group]:
+        raise Exception("Collection already exists")
+
+    data_root = get_data_root(group)
+    raw_data_files_root = data_root / f"{group}-{num}"
+    nexus_file_path = data_root / f"{group}-{num}.nxs"
```

Comment:
Nexus service currently has config for Nexus file extension, if we're going to also pass it from the visit service it should be because it is not hard coded and may change. I don't know if it ever will, so may be able to make this class a little shorter and just have data_files_root

Reply:
I think that's fine, tell services roughly where to put things then leave the details to them